strategy/strategy.go

```diff
@@ -0,0 +1,52 @@
+package strategy
+
+import (
+	"errors"
+	"reflect"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// Strategy describes a collection of metrics built with a struct, allowing any multi-level
+// combination of the 4 Prometheus metric types and other Strategies.
+// e.g. RED - describes a set of metrics that work well for monitoring
+// request-handling services.
+//
+//	type RED struct {
+//		Requests *prometheus.CounterVec
+//		Errors *prometheus.CounterVec
+//		Duration *prometheus.HistogramVec
+//	}
+type Strategy interface {
+	Register() error
+}
+
+// RegisterStrategyFields registers the fields of metrics and Strategies that
+// comprise a Strategy with the Prometheus DefaultRegisterer.
```

Comment:
I keep reading this comment and im not sure about it. How does this sound?

_The function "RegisterStrategyFields" is responsible for registering the metrics and fields associated with a strategy using the Prometheus DefaultRegisterer._

Reply:
@ignatospadov I'll work on this in a subsequent PR.